engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/WorkflowExecutionActor.scala

```diff
@@ -303,9 +304,11 @@ final case class WorkflowExecutionActor(workflowId: WorkflowId,
       goto(WorkflowExecutionFailedState) using stateData.mergeExecutionDiff(WorkflowExecutionDiff(Map(jobKey -> ExecutionStatus.Failed)))
     case Event(SucceededResponse(jobKey, returnCode, callOutputs), stateData) =>
       pushSuccessfulJobMetadata(jobKey, returnCode, callOutputs)
+      saveSuccessfulJobResults(jobKey, returnCode, callOutputs)
       handleJobSuccessful(jobKey, callOutputs, stateData)
     case Event(FailedNonRetryableResponse(jobKey, reason, returnCode), stateData) =>
       pushFailedJobMetadata(jobKey, returnCode, reason, retryableFailure = false)
+      saveUnsuccessfulJobResults(jobKey, returnCode, reason)
```

Comment:
I see you've also picked up #1005 - it might be possible to be clever here and not preserve failures for the job if the workflow is about to fail immediately.

This made me realise there might be a race condition if `RegisterWorkflowComplete` hits the `JobStoreActor` before a  `RegisterJobComplete` - and we could end up with job results left hanging around forever. This would be more likely that usual if the job failure and workflow failure both happen at the same time (i.e. like here...)